crypto/asn1/tasn_new.c

```diff
@@ -124,8 +124,12 @@ int asn1_item_embed_new(ASN1_VALUE **pval, const ASN1_ITEM *it, int embed)
                 goto memerr;
         }
         /* 0 : init. lock */
-        if (asn1_do_lock(pval, 0, it) < 0)
-            goto memerr2;
+        if (asn1_do_lock(pval, 0, it) < 0) {
+            if (!embed)
+                OPENSSL_free(*pval);
+            *pval = NULL;
```

Comment:
not sure, if that is OK for `embed = true`

Reply:
Ah, very good catch!